pkg/vgmanager/vgmanager_controller.go

```diff
@@ -413,3 +423,67 @@ func (r *VGReconciler) getMatchingDevicesForVG(volumeGroup *lvmv1alpha1.LVMVolum
 
 	return matchingDevices, delayedDevices, nil
 }
+
+func (r *VGReconciler) generateVolumeGroupNodeStatus(lvmdConfig *lvmdCMD.Config) (*lvmv1alpha1.LVMVolumeGroupNodeStatus, error) {
+
+	vgs, err := ListVolumeGroups(r.executor)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list volume groups. %v", err)
+	}
+
+	//lvmvgstatus := vgNodeStatus.Spec.LVMVGStatus
+	var statusarr []lvmv1alpha1.VGStatus
+
+	for _, vg := range vgs {
+		newStatus := &lvmv1alpha1.VGStatus{
+			Name:    vg.Name,
+			Reason:  "test",
+			Devices: vg.PVs,
+		}
+		statusarr = append(statusarr, *newStatus)
+	}
+
+	vgNodeStatus := &lvmv1alpha1.LVMVolumeGroupNodeStatus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.NodeName,
+			Namespace: r.Namespace,
+		},
+		Spec: lvmv1alpha1.LVMVolumeGroupNodeStatusSpec{
+			LVMVGStatus: statusarr,
+		},
+	}
+
+	return vgNodeStatus, nil
+}
+
+func (r *VGReconciler) updateStatus(ctx context.Context, lvmdConfig *lvmdCMD.Config) error {
+
+	vgNodeStatus, err := r.generateVolumeGroupNodeStatus(lvmdConfig)
+
+	nodeStatus := &lvmv1alpha1.LVMVolumeGroupNodeStatus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.NodeName,
+			Namespace: r.Namespace,
+		},
+	}
+
+	if err != nil {
+		r.Log.Error(err, "failed to generate nodeStatus")
+		return err
+	}
+
+	result, err := ctrl.CreateOrUpdate(ctx, r.Client, nodeStatus, func() error {
+		nodeStatus.Spec.LVMVGStatus = vgNodeStatus.Spec.LVMVGStatus
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "failed to create or update lvmvolumegroupnodestatus", "name", vgNodeStatus.Name)
+		return err
+	} else if result != controllerutil.OperationResultNone {
```

Comment:
any reason for not checking against `OperationResultUpdated`

Reply:
ack, I read the log `lvmvolumegroupnodestatus modified` and thought this condition is checking solely the modification.